src/main/java/Pony.java

```diff
@@ -0,0 +1,195 @@
+            }
+        } else if (action.equals("event")) {
+            // event expects description in command[1]
+            // check if command is valid
+            try {
+                if (commandSize != 2) {
+                    throw new PonyException.taskMissingError();
+                } else {
+                    // format of command: event <task> /at <time>
+                    String[] taskInfoArr = command[1].split(" /at ", 2);
+                    if (taskInfoArr.length != 2) {
+                        String format = "<task> /at <time>";
+                        throw new PonyException.taskFormatError(format);
+                    } else {
+                        String description = taskInfoArr[0];
+                        String time = taskInfoArr[1];
+                        Task newTask = new Event(description, time);
+                        this.tasks.add(newTask);
+                        System.out.println("Got it. I've added this task: " + newTask.toString());
+                        System.out.println("Now you have " + this.tasks.size() + " tasks in the list.");
+                    }
+                }
+            } catch (PonyException.taskMissingError | PonyException.taskFormatError e) {
+                System.out.println(e.getMessage());
+            } finally {
+                run();
+            }
+        } else if (action.equals("delete")) {
+            // delete expects input in command[1] -> which task to delete
+            // check if command is valid
+            try {
+                if (commandSize != 2) {
+                    throw new PonyException.taskMissingError();
+                } else {
+                    int taskIndex = Integer.parseInt(command[1]);
+                    Task target = this.tasks.get(taskIndex - 1);
+                    System.out.println("Noted. I've removed this task: ");
+                    System.out.println(target.toString());
+                    tasks.remove(taskIndex - 1);
+                    System.out.println("Now you have " + this.tasks.size() + " tasks in the list.");
+                }
+            } catch (PonyException.taskMissingError e) {
+                System.out.println(e.getMessage());
+            } catch (NumberFormatException e) {
+                System.out.println(new PonyException.taskInputError().getMessage());
+            } finally {
+                run();
+            }
+        } else if (action.equals("bye")) {
+            System.out.println(this.exit);
+        } else {
+            try {
+                throw new PonyException.inputError();
+            } catch (PonyException.inputError e) {
+                System.out.println(e.getMessage());
+            } finally {
+                run();
+            }
+        }
+    }
+
+    public void run() {
+        String[] command = takeCommand();
+        processCommand(command);
+    }
+
+    public static void main(String[] args) {
+        Pony myPony = new Pony();
```

Comment:
Since only one `Pony` object is created, why not make everything `static` instead?